pkg/webhook/resources/setting/validator.go

```diff
@@ -1451,6 +1460,60 @@ func (v *settingValidator) checkVlanStatusReady(config *networkutil.Config) erro
 	return nil
 }
 
+func (v *settingValidator) checkLeftoverPodIPs(oldSetting, newSetting *v1beta1.Setting) error {
+	if oldSetting.Value == newSetting.Value {
+		return nil
+	}
+
+	if oldSetting.Value != "" && newSetting.Value == "" {
+		return nil
+	}
+
+	if oldSetting.Value != "" && newSetting.Value != "" {
+		return nil
+	}
+
+	pods, err := v.podCache.List(util.LonghornSystemNamespaceName, labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	podNames := filterPodsByStorageNetwork(pods)
+	if len(podNames) > 0 {
+		return fmt.Errorf("the following pods: %s still contain IPs from previous storage network; clean them up before re-enabling", strings.Join(podNames, ", "))
+	}
+
+	return nil
+}
```

Comment:
Suggest making the syntax a bit clearer:


```suggestion
func (v *settingValidator) checkLeftoverPodIPs(oldSetting, newSetting *v1beta1.Setting) error {
	if oldSetting.Value == newSetting.Value {
		return nil
	}

	// Only check for leftover pod IPs when enabling the setting (transitioning from empty to non-empty)
	if !isSettingBeingEnabled(oldSetting.Value, newSetting.Value) {
		return nil
	}

	pods, err := v.podCache.List(util.LonghornSystemNamespaceName, labels.Everything())
	if err != nil {
		return err
	}

	podNames := filterPodsByStorageNetwork(pods)
	if len(podNames) > 0 {
		return fmt.Errorf("the following pods: %s still contain IPs from previous storage network; clean them up before re-enabling", strings.Join(podNames, ", "))
	}

	return nil
}

// isSettingBeingEnabled checks if a setting is transitioning from unset (empty) to set (non-empty).
// This is useful for validating conditions that should only apply when initially enabling a feature.
func isSettingBeingEnabled(oldValue, newValue string) bool {
	return oldValue == "" && newValue != ""
}
```

Reply:
Good suggestion, can change in a follow up, also didn't extract originally as I couldn't find other places where we need this check `isSettingBeingEnabled` so decided to extract when needed

Reply:
Done :heavy_check_mark: